siliconcompiler/core.py

```diff
@@ -4140,10 +4140,32 @@ def run(self):
                     # TODO: breakpoint logic:
                     # if task is breakpoint, then don't launch while len(running_tasks) > 0
 
+                    successful_deps = []
+                    deps_full_len = len(deps)
```

Comment:
As a small cleanup suggestion, it might be possible to make each of these booleans:

```python
dep_was_successful = False
had_deps = len(deps) > 0
```
(where `deps_was_successful` gets set to True in the for loop)

and then use those directly to tighten up the `if` statement below. If you think the information as you're currently storing it might be useful for other things in the future though, then feel free to disregard.